src/python/pants/core/util_rules/pants_environment_test.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Dict, List
+
+import pytest
+
+from pants.core.util_rules.pants_environment import PantsEnvironment
+
+
+@pytest.mark.parametrize(
```

Comment:
Very nice! I can't think of any missing edge case.

Reply:
Maybe more examples of the shorthand, like `B2`. It looks like we only test `B`.